frontend/model/captureLogs.js

```diff
@@ -0,0 +1,250 @@
+import sbp from '~/shared/sbp.js'
+import { CAPTURED_LOGS, SET_APP_LOGS_FILTER } from '~/frontend/utils/events.js'
+
+/*
+  - giConsole/[username]/limit - the limit of entries.
+  - giConsole/[username]/markerNth - the nth index in which marker log
+  - giConsole/[username]/count - total logs stored
+  - giConsole/[username]/lastEntry - latest log hash
+  - giConsole/[username]/markers - an array of markers at every nth entry
+*/
+
+// Configuration
+const ENTRIES_LIMIT = 5000
+const ENTRIES_MARKER_NTH = 1000
```

Comment:
@taoeffect. This file contains one of our most "sensitive" algorithms in the application for performance and security reasons. I strongly believe it should deserve its own unit tests. I can make it a priority on my tasks, and write unit tests to it _after_ we merge this PR. What do you think?

Reply:
If you would like to do that, sure, that's OK, but I think we should definitely save it for after this PR and move on to other things 👍

Reply:
Great, I'll add it to my todo list to do it only **after** this PR gets merged.